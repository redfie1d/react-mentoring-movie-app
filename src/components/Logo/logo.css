span {
  display: inline-flex;
  align-items: center;
  font-size: large;
  color: #f65261;
  cursor: pointer;
}
