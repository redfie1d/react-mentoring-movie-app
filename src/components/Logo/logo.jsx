import { useHistory } from 'react-router-dom';
import './logo.css';

const Logo = () => {
  const history = useHistory();

  const handleLogo = () => {
    history.push('/');
  };

  return (
    <span onClick={handleLogo}>
      <b>netflix</b>roulette
    </span>
  );
};

export default Logo;
