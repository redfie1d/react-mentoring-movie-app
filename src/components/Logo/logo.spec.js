import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import Logo from './logo';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('<Logo />', () => {
  it('should render without crashing', () => {
    const wrapper = mount(<Logo />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render Homepage when clicked', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Logo />
      </MemoryRouter>
    );
    wrapper.find('span').simulate('click');
    expect(mockHistoryPush).toHaveBeenCalledWith('/');
  });
});
