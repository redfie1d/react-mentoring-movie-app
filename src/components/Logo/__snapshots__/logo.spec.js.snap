// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render without crashing 1`] = `
"<Logo>
  <span onClick={[Function: handleLogo]}>
    <b>
      netflix
    </b>
    roulette
  </span>
</Logo>"
`;
